lib/src/int64_emulated.dart

```diff
@@ -0,0 +1,867 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: constant_identifier_names
+
+// Many locals are declared as `int` or `double`. We keep local variable types
+// because the types are critical to the efficiency of many operations.
+//
+// ignore_for_file: omit_local_variable_types
+
+import 'int32.dart';
+import 'int64.dart';
+import 'utilities.dart' as u;
+
+class Int64Impl implements Int64 {
```

Comment:
Agree with not using `Impl` anywhere.

If we worry about someone printing `int64.runtimeType`, we can even call it `Int64`, with a typedef for access (or just use a prefix, `impl.Int64`, in the abstract `Int64` superclass.)

Reply:
What's the problem with `Impl`?

Reply:
`Impl` is:
* An [abbreviation](https://dart.dev/effective-dart/design#avoid-abbreviations).
* Not adding anything meaningful (all non-abstract classes are implementations).
* Only there to avoid a name conflict, but Dart has other ways to do that: `_Foo` is enough for private types and `prefix.Foo` for public ones.
* Probably lots of other things, people have been discussing this for ages: https://www.google.com/search?q=avoid+impl+suffix

The "long names are long" guide says to only add words when they are needed to distinguish different things.
If there is no other concrete `Int64` class, it's already unique. You can distinguish it from the interface, but the interface probably shouldn't exist to begin with. The type should be a final and be the definition of what `Int64` is. No interface is needed, there will never be another implementation.

Reply:
> You can distinguish it from the interface, but the interface probably shouldn't exist to begin with

Yes, the interface is only there to avoid duplicating member documentation in implementation classes. You can look at the previous versions of the PR to see how that looked. It also caused an analyzer issue, but there may be ways to work around that issue without splitting this into an abstract + implementation class.

Reply:
The version without the abstract class: https://github.com/osa1/fixnum/blob/66768ce3037ad7703bc34c8f5fec1be279fda160/lib/src/int64.dart

Reply:
I just realized we don't need to document the JS implementation as `dart doc` runs generates documentation based on the native version. So I removed the abstract class again and moved all documentation comments to the native version. To work around the analyzer issue I just had to add one `ignore_for_file`.

Marking this as resolved as there's no longer an `Impl`.

Reply:
> I just realized we don't need to document the JS implementation as `dart doc` runs generates documentation based on the native version. 

native version or default configurable import? If the condition flips (e.g. using dart.library.ffi) does this change?

Reply:
It looks like it uses the default, with `export 'int64_emulated.dart' if (dart.library.ffi) 'int64_native.dart';` it uses documentation in `int64_emulated.dart`.